src/components/FilterBar/FilterDisplay/Inputs/LazyMultiSelectInput.react.js

```diff
@@ -0,0 +1,88 @@
+export class LazyMultiSelectInput extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state =  {
+      value: this.props.value === '' ? this.getDefaultValue() : this.props.value,
+      options: []
+    }
+  }
+
+  componentDidMount() {
+    let filter = this.getFilterFromFilterBarStore();
+    this.setState({ options: [] });
+    filter.value = this.state.value;
+  }
+
+  getFilterFromFilterBarStore() {
+    return(this.context.filterBarStore.getFilter(this.props.filterUid));
+  }
+
+  componentDidUpdate() {
+    let multiSelectInput = $(React.findDOMNode(this.refs.reactLazyMultiSelect));
+    let filter = this.getFilterFromFilterBarStore();
+    multiSelectInput.select2({
+      minimumInputLength: 3,
+      multiple: true,
+      ajax: {
+        url: filter.url,
+        quietMillis: 250,
+        dataType: 'json',
+        data: function (term, page) {
+          return {
+            q: term
+          };
+        },
+        results: function (data, params) {
+          return {
+            results: $.map(data, function (item) {
+              return {
+                text: item.label,
+                id: item.value
+              }
+            })
+          };
+        }
+      }
+    });
+    multiSelectInput.on('change', this.onSelect.bind(this));
+  }
+
+  componentWillUnmount() {
+    if (this.serverRequest !== undefined) {
+      this.serverRequest.abort();
+    }
+  }
+
+  getDefaultValue() {
+    let filter = this.getFilterFromFilterBarStore();
+    return([filter.default]);
+  }
+
+  onSelect(event) {
+    let filter = this.getFilterFromFilterBarStore();
+    filter.value = event.target.value;
+  }
+
+  render() {
+    return (
+      <li>
+        <input
+          className="form-control"
+          value={this.state.value}
+          ref="reactLazyMultiSelect"
```

Comment:
It's currently working without it. See below. Regarding saved search, I'm not sure how it is   done either, I suppose it must be tested within Neptune. Right.

![lazy_filters](https://cloud.githubusercontent.com/assets/12091869/23935653/17ed4c8e-09a1-11e7-8591-360ad6d8c9ef.gif)